From 4453f965d9266a0d57cb9cad73df1755ad4acf71 Mon Sep 17 00:00:00 2001
From: Wojciech Bartczak <wbartczak@marvell.com>
Date: Thu, 17 Jun 2021 16:59:14 -0700
Subject: [PATCH 1590/1921] driver: Fixes conversion for
 BOARD-MAC-ADDRESS-ID-NUM in octeontx_info

This change fixes conversion used to convert BOARD-MAC-ADDRESS-ID-NUM
from string to binary number. Previously it was base 10 number,
now it matches EBF implementation and it's converted as base 16.

Signed-off-by: Wojciech Bartczak <wbartczak@marvell.com>
Change-Id: Id0ab29cfe38561c641530de2f28aaaa52592d092
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/54320
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/soc/marvell/octeontx_info.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/soc/marvell/octeontx_info.c b/drivers/soc/marvell/octeontx_info.c
index bb487419c95f..4e3142503173 100644
--- a/drivers/soc/marvell/octeontx_info.c
+++ b/drivers/soc/marvell/octeontx_info.c
@@ -89,7 +89,7 @@ static int oct_brd_proc_show(struct seq_file *seq, void *v)
 			seq_printf(seq, "mac_addr_count: %s\n",
 				   brd.board_num_of_mac_id);
 
-		if (!kstrtou32(brd.board_num_of_mac_id, 10, &num)) {
+		if (!kstrtou32(brd.board_num_of_mac_id, 16, &num)) {
 			seq_printf(seq, "mac_addr_count: %s\n",
 				   brd.board_num_of_mac_id);
 
-- 
2.31.1

